src/coreclr/jit/gentree.cpp

```diff
@@ -1603,6 +1603,36 @@ CallArg* CallArgs::GetArgByIndex(unsigned index)
     return cur;
 }
 
+//---------------------------------------------------------------
+// GetUserArgByIndex: Get an argument with the specified index.
+//   Unlike GetArgByIndex, this function ignores non-user args
+//   like r2r cells.
+//
+// Parameters:
+//   index - The index of the argument to find.
+//
+// Returns:
+//   A pointer to the argument.
+//
+// Remarks:
+//   This function assumes enough arguments exist.
+//
+CallArg* CallArgs::GetUserArgByIndex(unsigned index)
+{
+    CallArg* cur = m_head;
+    for (unsigned i = 0; i < index || cur->IsArgAddedLate();)
+    {
+        if (!cur->IsArgAddedLate())
+        {
+            i++;
+        }
+        assert((cur != nullptr) && "Not enough arguments in GetArgByIndex");
+        cur = cur->GetNext();
+    }
+
+    return cur;
+}
```

Comment:
This will hit null ref before the assert, and looks like the assert message should be updated.

`CallArg::IsArgAddedLate` is maybe not actually the thing we want to be using here either, it will return false for ret buf and type parameters but I'm not sure we would call those "user" args. I think it's ok, but maybe add it to the remarks?